### ISSUE TITLE
New Pangeo deployment worker-pool won't autoscale to run dask worker pods

### ISSUE THREAD

**Issue by mwengren** [pangeo-data/pangeo#431]
This runs counter to some other OS projects I've been involved with in terms of asking for help on GitHub issues rather than a mailing list, but I think I'm following the [rules](http://pangeo.io/contact.html) in posting this, so here goes:
 
I have some cloud credits at my disposal and am trying to set up a vanilla Pangeo environment following the [cloud install docs](http://pangeo.io/setup_guides/cloud.html).

I made a few minor customizations to my deployment, but nothing too major to diverge from a typical GCP Pangeo install based on the docs AFAIK.  The only changes I made were:

- HTTPS via LetsEncrypt
- auth using Google Oauth (both OAuth and HTTPS following the zero2Jupyterhub docs).  
- reduced the initial  size of the `worker-pool` node pool by changing `MAX_WORKER_NODES=10` as a (possibly naive) attempt to limit costs to start.  

My issue is that the `worker-pool` node pool will not autoscale and running several of the example notebooks results in many `dask-USERNAME-*****` pods being requested but all coming back with errors such as:

```
PodUnschedulable Cannot schedule pods: Insufficient cpu.
PodUnschedulable Cannot schedule pods: Insufficient memory.
```

I went back and increased the worker-pool `--max-nodes` size to 100 and updated my cluster by running the  `gcloud container clusters update` command:
```
gcloud container clusters update $CLUSTER_NAME --zone=$ZONE --node-pool=worker-pool --enable-autoscaling --max-nodes=$MAX_WORKER_NODES --min-nodes=$NUM_WORKER_NODES
```
However the worker-pool never spins up any instances despite autoscaling being set properly etc.

I had couple of questions:
- anyone know what is wrong with my setup and have a quick fix (I hope)?
- how is it determined which node pool the dask worker pods are requested in (is there an explicit configuration allocating them in `worker-pool` and if so where?).  I wasn't able to find it digging through various config files I could find and also in the GCP console pod metadata.  I found the `worker-template.yaml` file in my running JupyterHub environment, but didn't see anything relevant. 

For reference, my unchanged worker-template.yaml looks like this:
```
metadata:
spec:
  restartPolicy: Never
  containers:
  - args:
      - dask-worker
      - --nthreads
      - '2'
      - --no-bokeh
      - --memory-limit
      - 6GB
      - --death-timeout
      - '60'
    image: pangeo/notebook:92c4bfd
    name: dask-worker
    securityContext:
      capabilities:
        add: [SYS_ADMIN]
      privileged: true
    env:
      - name: GCSFUSE_BUCKET
        value: pangeo-data
    resources:
      limits:
        cpu: "1.75"
        memory: 6G
      requests:
        cpu: "1.75"
        memory: 6G
```

These may be beginner Kubernetes questions, but this is my first foray into Kubernetes.  Thanks for any help.  I would love to get this running properly.  It seems either I misconfigured something or the out of the box setup isn't working properly.  I'm sure with more troubleshooting, I can figure it out but until then I figured I'd ask here.

I will say that both the Pangeo docs and the zero2Jupyterhub docs have been great learning tools and you all deserve credit for investing time in good documentation for new users like myself - thanks!  Up until this point it has been super easy and fun.

Hopefully I can get over the finish line.  Regardless, these docs are great Kubernetes 101 resources so much appreciated.

**Comment by rabernat** [comment]
Welcome! Feel free to come by our weekly check in tomorrow to discuss your issue: https://github.com/pangeo-data/pangeo/issues/427

**Comment by mwengren** [comment]
So I resolved this, I think my error was trying to use too small an instance size for the `worker-pool` - I copied from the default pool and used n1-standard-2, which may be too small to run the dask workers.  Wiped out the worker-pool and re-ran the commands with n1-standard-4 size instances and dask workers started properly.   

I have a new issue however, my jupyterhub session won't restart no matter what I do in the client (cache clear, refresh, etc).  No pod initializes in the cluster.  If I can't troubleshoot that will open another ticket.  

Will try to make a future meeting but the 11 AM time today I may not make, thanks.
